core/include/jml/model.hpp

```diff
@@ -57,14 +57,20 @@ class Model {
 public:
     Model();
     std::unique_ptr<Vector> apply(Vector& in);
+    // Adds testing data using iterators, for the start and end locations of
+    // the wanted input and output data lists.
+    void add_testing_data (
+        std::vector<Vector>::iterator inb, std::vector<Vector>::iterator ine,
+        std::vector<Vector>::iterator otb, std::vector<Vector>::iterator ote
```

Comment:
nit: I would prefer more verbose argument names, personally.

Reply:
I put a comment up above which hopefully explains it somewhat.